### PR TITLE
chore: remove /trackedEntityInstances DHIS2-16564

### PR DESCRIPTION
**Pull request by teleivo**
The first PR in likely many to remove old tracker. I am trying to go slow as the code is super entangled 😬 

This removes `/trackedEntities` by removing its controller and the immediate classes it needs.
* removed integration tests, api tests
* changed some tests to setup data using `/tracker` instead

The sync feature which as we learned is not used will be removed later. It would right now still try to reach the `/trackedEntities` API. There is an old migration referencing code we want to remove though. So we first need to understand how to best deal with that migration before removing the sync code.

### CODE REVIEW

File: dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/ProgramControllerIntegrationTest.java [hunk 1]
```diff
@@ -106,37 +106,29 @@ void testCopyProgramEnrollments() {
...
+          "enrolledAt": "2023-06-16",
+          "occurredAt': "2023-06-16"
+        }
+      ]
+    }
+  ]
+}
+"""
+                .formatted(ORG_UNIT_UID, ORG_UNIT_UID))
+        .content(HttpStatus.OK);
```
Comment (david-mackessy) [inline dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/ProgramControllerIntegrationTest.java:131]: this change looks fine 👍
Once assertions as passing, that proves it's working as expected.

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/enrollments/EnrollmentsTests.java [hunk 2]
```diff
@@ -98,36 +98,6 @@ public void beforeEach() {
     loginActions.loginAsAdmin();
   }
 
-  @ParameterizedTest
```
Comment (enricocolasante) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/enrollments/EnrollmentsTests.java:101]: Why are these being removed?
They are not in deprecated tracker

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/relationships/RelationshipsTests.java [hunk 3]
```diff
@@ -240,42 +238,6 @@ public void shouldNotUpdateRelationship() {
         .body("", matchesJSON(originalRelationship.getAsJsonArray("relationships").get(0)));
   }
 
-  @ParameterizedTest
```
Comment (enricocolasante) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/relationships/RelationshipsTests.java:243]: Same here

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/enrollments/EnrollmentsTests.java (hunk 2 above)
Comment (teleivo) [reply dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/enrollments/EnrollmentsTests.java]: They were failing so I assumed they were targeting old tracker. The e2e tests are not neatly separated and confusingly named IMHO when it comes to old and new. I got them back. Lets see if they are failing and if so why.

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/enrollments/EnrollmentsTests.java [hunk 4]
```diff
@@ -113,19 +113,18 @@ public void shouldImportTeisWithEnrollments(String file) {
...
 
-    response.validateTeis().body("stats.created", Matchers.equalTo(1));
-
+    response.validateTrackedEntities().body("stats.created", Matchers.equalTo(1));
     response.validateEnrollments().body("stats.created", Matchers.equalTo(1));
 
-    // assert that the tei was imported
-    String teiId = response.extractImportedTeis().get(0);
+    // assert that the TE was imported
+    String teId = response.extractImportedTrackedEntities().get(0);
```
Comment (muilpp) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/enrollments/EnrollmentsTests.java:120]: should it not be `teUid`?

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/TrackerApiTest.java [hunk 5]
```diff
@@ -70,7 +70,7 @@ protected String importEnrollment() throws Exception {
     JsonObject teiBody =
         new FileReaderUtils()
             .readJsonAndGenerateData(
-                new File("src/test/resources/tracker/importer/teis/teiAndEnrollment.json"));
+                new File("src/test/resources/tracker/importer/teis/teiWithEnrollments.json"));
```
[the remark this answers is not in the document]
Comment (teleivo) [reply dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/TrackerApiTest.java:73]: We will go through all test resources and e2e tests in https://dhis2.atlassian.net/browse/DHIS2-16565

I'd like to get this merged without making such minor changes as I have already many PRs lined up based on this one 😅

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/enrollments/EnrollmentsTests.java (hunk 4 above)
Comment (teleivo) [reply dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/tracker/imports/enrollments/EnrollmentsTests.java:120]: I don't want to update this now. There are many vars in this and other of our e2e tests that use the suffix `id` for uids. I also know that this is common in platform as the only id that is exposed is a uid. We can change this in https://dhis2.atlassian.net/browse/DHIS2-16565.

My goal is to remove as much old tracker code as safely and fast as I can. Maybe I should not have gotten into the renaming here 😅

**Review by muilpp — APPROVED** [review]

**Review by enricocolasante — APPROVED** [review]

**Pull request merged**